amun/api_v1.py

```diff
@@ -109,137 +109,313 @@ def _adjust_default_requests(dict_: dict) -> None:
+                "parameters": parameters,
+            },
+            202,
+        )
+
+    pod_logs: List[str] = []
+    try:
+        pod_ids: List[str] = _OPENSHIFT._get_pod_ids_from_job(inspection_id, Configuration.AMUN_INSPECTION_NAMESPACE)
+
+        for pod_id in pod_ids:
+            log: str = _OPENSHIFT.get_pod_log(pod_id, namespace=Configuration.AMUN_INSPECTION_NAMESPACE)
+            pod_logs.append(log)
+    except NotFoundException as exc:
+        return (
+            {
+                "error": "No pods for the given inspection id was not found",
+                "status": inspection_status,
+                "parameters": parameters,
+            },
+            404,
+        )
+
+    inspection_logs: List[Dict[str, Any]] = []
+    for pod, pod_log in zip(pod_ids, pod_logs):
+        log: Dict[str, Any]
+        try:
+            log = json.loads(pod_log)
+        except json.JSONDecodeError:
+            _LOGGER.exception("Failed to parse log from pod %s: %r", pod, pod_log)
+            continue
+
+        inspection_logs.append(log)
+
+    if not any(inspection_logs):
+        _LOGGER.error("Inspection run did not produce any logs or it was deleted by OpenShift")
+        return (
+            {
+                "error": "Inspection run did not produce any logs or it was deleted by OpenShift",
+                "status": inspection_status,
+                "parameters": parameters,
+            },
+            404,
+        )
+
+    return {"logs": inspection_logs, "parameters": parameters}, 200
 
 
 def get_inspection_build_log(inspection_id: str) -> tuple:
     """Get build log of an inspection."""
-    parameters = {'inspection_id': inspection_id}
+    parameters = {"inspection_id": inspection_id}
 
     try:
-        status = _OPENSHIFT.get_pod_log(
-            inspection_id + '-1-build',
-            Configuration.AMUN_INSPECTION_NAMESPACE
-        )
+        status = _OPENSHIFT.get_pod_log(inspection_id + "-1-build", Configuration.AMUN_INSPECTION_NAMESPACE)
     except NotFoundException:
-        return {
-            'error': 'Build log with for the given inspection id was not found',
-            'parameters': parameters
-        }, 404
+        return (
+            {"error": "Build log with for the given inspection id was not found", "parameters": parameters},
+            404,
+        )
 
-    return {
-        'log': status,
-        'parameters': parameters
-    }, 200
+    return {"log": status, "parameters": parameters}, 200
 
 
 def get_inspection_status(inspection_id: str) -> tuple:
     """Get status of an inspection."""
-    parameters = {'inspection_id': inspection_id}
+    parameters = {"inspection_id": inspection_id}
+
+    workflow_status = None
+    try:
+        wf: Dict[str, Any] = _OPENSHIFT.get_workflow(
+            label_selector=f"inspection_id={inspection_id}", namespace=_OPENSHIFT.amun_inspection_namespace,
+        )
+        workflow_status = wf["status"]
```

Comment:
This will be handy in thoth-common, so we can use it on metrics-exporter.

Reply:
Do we want to have a specific function for it in Thoth common? I.e. `get_workflow_status`?

Reply:
I think it would be nice to have it :+1: